lib/p2psocket.js

```diff
@@ -77,93 +101,200 @@ P2PSocket.prototype.parseInv = function(payload) {
     var offset = size + i * 36;
     content.inventory.push({
       type: payload.readUInt32LE(offset),
-      hash: payload.slice(offset + 4, offset + 4 + 32)
+      hash: inPlaceReverse(payload.slice(offset + 4, offset + 4 + 32)).toString('hex')
     });
   }
 
   return content;
 }
 
+P2PSocket.prototype.parseBlock = function(payload) {
+
+  var header = payload.slice(0,80);
+
+  var sha256_pass1 = crypto.createHash('sha256').update(header).digest();
+  var blockhash = inPlaceReverse(crypto.createHash('sha256').update(sha256_pass1).digest()).toString('hex');
+
+  var content = {
+    hash:              blockhash,
+    confirmations:     undefined,
+    size:              payload.length,
+    height:            undefined,
+    version:           payload.readUInt32LE(0),
+    merkleroot:        inPlaceReverse(payload.slice(36,68)).toString('hex'),
+    tx:                [],
+    time:              payload.readUInt32LE(68),
+    nonce:             payload.readUInt32LE(76),
+    bits:              inPlaceReverse(payload.slice(72,76)).toString('hex'),
+    difficulty:        undefined,
```

Comment:
I stared at that page for hours and couldn't figure out the math to get it from the bits.